treeherder/etl/resultset_loader.py

```diff
@@ -31,8 +31,11 @@ def process(self, message_body, exchange):
             store_result_set_data(repo, [transformed_data])
 
         except ObjectDoesNotExist:
+            repo_info = (message_body["details"]
+                         if "details" in message_body
+                         else message_body["payload"])
```

Comment:
Which do you prefer out of the above, or:
```python
repo_info = message_body.get("details", message_body["payload"])
```

Also I know this is pre-existing, but I think the `ObjectDoesNotExist` should just be around the `Repository.objects.get()`? Plus we're filtering by respository `url`, but there's no index on it. Could you file some bugs for following up on these?